src/Agent.Worker/TaskRunner.cs

```diff
@@ -63,6 +63,29 @@ public async Task RunAsync()
             ArgUtil.NotNull(ExecutionContext, nameof(ExecutionContext));
             ArgUtil.NotNull(ExecutionContext.Variables, nameof(ExecutionContext.Variables));
             ArgUtil.NotNull(Task, nameof(Task));
+
+            bool logTaskNameInUserAgent = AgentKnobs.LogTaskNameInUserAgent.GetValue(ExecutionContext).AsBoolean();
+
+            if (logTaskNameInUserAgent)
+            {
+                VssUtil.PushTaskIntoAgentInfo(Task.Name ?? "", Task.Reference?.Version ?? "");
```

Comment:
Nit: If we don't have task name/versions, can we place smth e.g. "Unknown" for better visibility?

Reply:
good suggestion but we should be able to deal with empty strings just as well.  If both Task and Version is null, it should appear in UA (iirc) as (Task:-)